### PR TITLE
Fix typings generation, update API references and CT Composables pages

### PR DESCRIPTION
**Pull request by filipsobol**
This PR:
- adds API Reference for:
  - `@vue-storefront/commercetools`,
  - `@vue-storefront/cache`,
  - `@vue-storefront/middleware`,
- updates Composables pages in commercetools documentation,
- fixes types generation for `@vue-storefront/cache` and `@vue-storefront/middleware`.

### CODE REVIEW

File: packages/core/cache/api-extractor.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "../../api-extractor.base.json",
+  "mainEntryPointFilePath": "./lib/index.d.ts",
+  "dtsRollup": {
+    "untrimmedFilePath": "./lib/<unscopedPackageName>.d.ts"
+  },
+  "docModel": {
+    "apiJsonFilePath": "<projectFolder>/core/docs/core/api-reference/<unscopedPackageName>.api.json"
+  }
+}
```
Comment (lukaszjedrasik) [inline packages/core/cache/api-extractor.json:10]: Missing newline at the end of the file.

```suggestion
}

```

File: packages/core/middleware/api-extractor.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "../../api-extractor.base.json",
+  "mainEntryPointFilePath": "./lib/index.d.ts",
+  "dtsRollup": {
+    "untrimmedFilePath": "./lib/<unscopedPackageName>.d.ts"
+  },
+  "docModel": {
+    "apiJsonFilePath": "<projectFolder>/core/docs/core/api-reference/<unscopedPackageName>.api.json"
+  }
+}
```
Comment (lukaszjedrasik) [inline packages/core/middleware/api-extractor.json:10]: Missing newline at the end of the file.

```suggestion
}

```

File: packages/commercetools/api-client/src/api-extractor-data.ts [hunk 3]
```diff
@@ -1,4 +1,15 @@
 import * as apiMethods from './api';
 
+/**
+ * `api-client` for commercetools integration for Vue Storefront 2.
+ *
+ * @remarks
+ * The `@vue-storefront/commercetools-api` library includes everything needed to fetch data from the
+ * commercetools eCommerce platform. This includes API client configuration, API  endpoints and
+ * GraphQL types and fragments.
```
Comment (igorwojciechowski) [inline packages/commercetools/api-client/src/api-extractor-data.ts:9]:
```suggestion
 * The `@vue-storefront/commercetools-api` library includes everything needed to fetch data from the
 * commercetools eCommerce platform. This includes API client configuration, API endpoints, and
 * GraphQL types and fragments.
```

**Review by lukaszjedrasik — APPROVED** [review]
Great job 🚀 
Just two small suggestions from my side.

File: packages/commercetools/composables/src/index.ts [hunk 4]
```diff
@@ -3,6 +3,17 @@ import { track } from '@vue-storefront/core';
 
 track('VSFCommercetools');
 
+/**
+ * `composables` for commercetools integration for Vue Storefront 2.
+ *
+ * @remarks
+ * The `@vue-storefront/commercetools-api` library includes everything needed to fetch data from the
+ * Server Middleware and display them to in agnostic and formatted form. This includes composables
+ * and getters.
```
Comment (igorwojciechowski) [inline packages/commercetools/composables/src/index.ts:12]:
```suggestion
 * The `@vue-storefront/commercetools-api` library includes everything needed to fetch data from the
 * Server Middleware and display them in agnostic and formatted form. This includes composables
 * and getters.
```

File: packages/core/docs/.vuepress/config.js [hunk 5]
```diff
@@ -164,29 +164,29 @@ module.exports = {
...
-            ['/commercetools/composables/use-make-order', 'useMakeOrder'],
+            ['/commercetools/composables/use-facet', 'useFacet'],
             ['/commercetools/composables/use-forgot-password', 'useForgotPassword'],
-						['/commercetools/composables/use-store', 'useStore']
+            ['/commercetools/composables/use-make-order', 'useMakeOrder'],
+            ['/commercetools/composables/use-product', 'useProduct'],
+            ['/commercetools/composables/use-review', 'useReview'],
+            ['/commercetools/composables/use-shipping-provider', 'useShippingProvider'],
+            ['/commercetools/composables/use-shipping', 'useShipping'],
+						['/commercetools/composables/use-store', 'useStore'],
```
Comment (igorwojciechowski) [inline packages/core/docs/.vuepress/config.js:177]:
```suggestion
['/commercetools/composables/use-store', 'useStore'],
```

**Review by igorwojciechowski — APPROVED** [review]

**Pull request merged**